### PR TITLE
Core/Logging: rework logging to POCO::Logging

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
#### CHANGES PROPOSED:
- Title
- This logging looks like in TC

#### TESTS PERFORMED:
- Win in game
- Win build (VS 2019 16.1.3)

#### HOW TO TEST THE CHANGES:
1. Apply this PR :/
2. Change config options in `LogChannel`(Times,Pattern,Optional1,Optional2,Optional3,Optional4) and `Logger` (LogLevel)

#### KNOWN ISSUES AND TODO LIST:
- [x] Bad encoding in Windows Console
- [ ] Split logs into categories
- [x] DB logging
- [ ] Check average diff
- [ ] Check dir in Unix

#### Target branch(es):
Master

#### About POCO
https://pocoproject.org
https://pocoproject.org/docs/package-Foundation.Logging.html

#### About fmt
https://github.com/fmtlib/fmt

#### Other
I set all log filter `root` for testing all messages

### Difference between old and new
1. Now each message belongs to its own category (logger). Like TC logging
2. Log levels - 8. Before - 3. 
4. Maximum customization of message output. See formats (Type,Times,Pattern,Optional1,Optional2,Optional3,Optional4) for `LogChannel` in config file

**Example:** `Pattern`
- The default for type `Console` - `%t`
image
- If change for example to `%s - %t`
image
- Or `%H:%M:%S %t`
image

### CODE REVIEW

**Comment by BarbzYHOOL** [comment]
To test, check the settings in worldserver.conf.dist  around `LOGGING SYSTEM SETTINGS`

File: .travis.yml [hunk 1]
```diff
@@ -71,6 +71,6 @@ jobs:
         - source ./apps/ci/ci-before_install.sh
       install:
         - source ./apps/ci/ci-install.sh ON
-        - source ./apps/ci/ci-install-modules.sh
+        # - source ./apps/ci/ci-install-modules.sh For test no need module
```
Comment (BarbzYHOOL) [inline .travis.yml]: ok but don't forget to put them back o_0

Comment (Winfidonarleyan) [reply .travis.yml]: Yes yes, hah

**Comment by Winfidonarleyan** [comment]
Hmm, error at startup console
```ini
Loading GameObject models...
Unable to open 'GameObjectModels.dtree' file.
```

**Comment by Winfidonarleyan** [comment]
Added examples and difference between old and new

**Comment by BarbzYHOOL** [comment]
the timestamp on the left would be very useful

File: apps/ci/ci-worldserver-dry-run.sh [hunk 2]
```diff
@@ -6,7 +6,7 @@ if [ "$TRAVIS_BUILD_ID" = "1" ]
 then
   echo "start worldserver dry-run"
   git clone --depth=1 --branch=master --single-branch https://github.com/ac-data/ac-data.git /home/travis/build/azerothcore/azerothcore-wotlk/env/dist/data
-  sed -e "s/;;acore_auth/;$DB_RND_NAME;auth_$DB_RND_NAME/" -e "s/;;acore_world/;$DB_RND_NAME;world_$DB_RND_NAME/" -e "s/;;acore_characters/;$DB_RND_NAME;characters_$DB_RND_NAME/" ./data/travis/worldserver.conf >./env/dist/etc/worldserver.conf
+  sed -e "s/;;acore_auth/;$DB_RND_NAME;auth_$DB_RND_NAME/" -e "s/;;acore_world/;$DB_RND_NAME;world_$DB_RND_NAME/" -e "s/;;acore_characters/;$DB_RND_NAME;characters_$DB_RND_NAME/" ./env/dist/etc/worldserver.conf.dist >./env/dist/etc/worldserver.conf
```
Comment (BarbzYHOOL) [inline apps/ci/ci-worldserver-dry-run.sh]: ?

Comment (Winfidonarleyan) [reply apps/ci/ci-worldserver-dry-run.sh]: For test

**Comment by Winfidonarleyan** [comment]
Hm, strange Travis. The option is disabled :/
```
Enabling database logging...
```

And 
```
./apps/ci/ci-worldserver-dry-run.sh: line 12: 15096 Segmentation fault      (core dumped) ./env/dist/bin/worldserver --dry-run
```

**Comment by Winfidonarleyan** [comment]
Need test in Lunix default and at `dry-run`

**Comment by Hir0shi** [comment]
@Winfidonarleyan 
please stop all updates, whit https://github.com/azerothcore/azerothcore-wotlk/issues/1997

**Comment by Winfidonarleyan** [comment]
@Hir0shi Do you have any confirmation that my PR is bad for RAM?

**Comment by Hir0shi** [comment]
@Winfidonarleyan 
NO, But Azerothcore A big problem!

please help to fixed!

ram and cpu will to usage full !

**Comment by poszer** [comment]
> @Hir0shi Do you have any confirmation that my PR is bad for RAM?

He don't know what he saying. Just focus on what are you doing @Winfidonarleyan

**Comment by poszer** [comment]
> @Winfidonarleyan
> NO, But Azerothcore A big problem!
> 
> please help to fixed!
> 
> ram and cpu will to usage full !

Will you stop? You open issue and wait!

**Comment by Winfidonarleyan** [comment]
@Hir0shi This problem has been known for a long time and I don't know how to fix it

**Comment by Hir0shi** [comment]
OK, baby

**Comment by poszer** [comment]
> @poszer
> iwant just help!
> OK, baby

How the fck you gona help, if you posting XXXXXX Times same text. You have opened issue, explained problem, and we saw it. You don't need to write it XX times, Just wait, when someone find the way to fix it, and when someone of us get some free time it will be fixed if there is actually problem. Understand one thing, this isn't our job, we are all working here for free, we have family and we have life without this!

**Comment by BarbzYHOOL** [comment]
I might test it in the future on linux, but can't at the moment.

**Comment by Winfidonarleyan** [comment]
I maybe rewrite this pr, later

**Comment by BarbzYHOOL** [comment]
ok i might be able to test it then, i'm almost done reinstalling everything and improved logging is a nice thing to have !

**Comment by BarbzYHOOL** [comment]
i wanted this :(

**Comment by Winfidonarleyan** [comment]
This very good changes

**Comment by BarbzYHOOL** [comment]
> This very good changes

??? but we don't have it :(

**Comment by FrancescoBorzi** [comment]
why closing then? can you open a new one?

**Comment by Winfidonarleyan** [comment]
Yeah, I'll make a new one soon.

**Pull request closed without merging**